File: src/main/java/github/macro/build_info/gems/GemBuild.java
package github.macro.build_info.gems;

import java.util.List;
import java.util.Optional;

/**
 * Created by Macro303 on 2020-Jan-22.
 */
public class GemBuild {
	private final List<List<Optional<Gem>>> links;
	private final List<UpdateGem> updates;

	public GemBuild(List<List<Optional<Gem>>> links, List<UpdateGem> updates) {
		this.links = links;
		this.updates = updates;
	}

	public List<List<Optional<Gem>>> getLinks() {
		return links;
	}

	public List<UpdateGem> getUpdates() {
		return updates;
	}

	@Override
	public String toString() {
		return "GemBuild{" +
				"links=" + links +
				", updates=" + updates +
				'}';
	}
}

File: src/main/kotlin/github/macro/build_info/gems/GemBuild.kt
package github.macro.build_info.gems

import javafx.beans.property.SimpleListProperty
import javafx.collections.FXCollections
import tornadofx.getValue
import tornadofx.setValue

/**
 * Created by Macro303 on 2020-Jan-23.
 */
class GemBuild(
    links: List<List<Gem?>>,
    updates: List<UpdateGem>
) {
    val linksProperty = SimpleListProperty<List<Gem?>>()
    var links by linksProperty

    val updatesProperty = SimpleListProperty<UpdateGem>()
    var updates by updatesProperty

    init {
        this.links = FXCollections.observableList(links)
        this.updates = FXCollections.observableList(updates)
    }

    override fun toString(): String {
        return "GemBuild(links=$links, updates=$updates)"
    }
}